packages/block-library/src/embed/embed-no-preview.native.js

```diff
@@ -118,7 +115,7 @@ const EmbedNoPreview = ( { label, icon, isSelected, onPress } ) => {
 				onPress={ onPressContainer }
 			>
 				<View style={ containerStyle }>
-					<Icon icon={ icon } fill={ embedIconStyle.fill } />
+					<BlockIcon icon={ icon } />
```

Comment:
Now that we've enabled all the variations, icons can also be objects:
https://github.com/WordPress/gutenberg/blob/3e77c785d114e3a4cad7d34b5df66a0fbea6f1f9/packages/block-library/src/embed/icons.js#L26-L35
`BlockIcon` component was used in the web version and it handles this case as well.